src/codegen/doc/mod.rs

```diff
@@ -503,7 +513,28 @@ where
             )?;
         }
 
-        for parameter in fn_.parameters() {
+        let parameters: Vec<_> = fn_
+            .parameters()
+            .iter()
+            .filter(|param| {
+                let is_array_length_param = if param.name.len() > 3 {
+                    let array_param_name = &param.name[2..];
+                    &param.name[0..2] == "n_"
+                        && fn_.parameters().iter().any(|p| p.name == array_param_name)
+                } else {
+                    false
+                };
+
+                !IGNORED_C_FN_PARAMS.contains(&param.name.as_str())
```

Comment:
I feel like this is easier to read if you apply De Morgan's law and turn this into a `!(... || ...)`, perhaps even a:


```rust
let ignore = IGNORED_C_FN_PARAMS.contains(&param.name.as_str())
    || param.direction != ParameterDirection::Out
    || ...;
!ignore
```